internal/auth/permissions.go

```diff
@@ -0,0 +1,99 @@
+package auth
+
+import (
+	"net/http"
+	"strings"
+)
+
+const (
+	Forbidden                      = "Forbidden"
+	AuthorizationServerUnavailable = "Authorization server unavailable"
+	BadRequest                     = "Unable to verify request"
+)
+
+func CreatePermissionsMiddleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			// /api, /api/v1 and "auth" don't require permissions
+			parts := strings.Split(r.URL.Path, "/")
+			if len(parts) <= 2 ||
+				parts[2] == "auth" {
+				next.ServeHTTP(w, r)
+				return
+			}
+
+			// Extract resource and action from the request
+			resource, action := extractResourceAndAction(r)
+			if resource == "" || action == "" {
+				http.Error(w, BadRequest, http.StatusServiceUnavailable)
```

Comment:
I would say to log it.  It might mean that the extraction logic failed to find the resource and action

Reply:
done